cucoslib/workers/recommender.py

```diff
@@ -61,13 +64,21 @@ def str_value_cleaner(safe_string):
 
     def execute_gremlin_dsl(self, payload):
         """Execute the gremlin query and return the response."""
-        response = requests.post(
-            self._bayesian_graph_url, data=json.dumps(payload))
-        json_response = response.json()
-        if response.status_code != 200:
+        try:
+            s = requests.Session()
+            a = requests.adapters.HTTPAdapter(max_retries=int(os.getenv('MAX_GREMLIN_RETRY_COUNT', '3')))
```

Comment:
I think this won't retry on 500. You need to pass instance of `Retry` class instead of just `int`. See from example here: https://www.peterbe.com/plog/best-practice-with-retries-with-requests (search for `status_forcelist`).